redfish-core/lib/processor.hpp

```diff
@@ -1159,6 +1184,106 @@ inline void handleProcessorCollectionHead(
         "</redfish/v1/JsonSchemas/ProcessorCollection/ProcessorCollection.json>; rel=describedby");
 }
 
+inline void
+    setProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                     const dbus::utility::MapperServiceMap& serviceMap,
+                     const std::string& objectPath,
+                     bool locationIndicatorActive)
+{
+    for (const auto& [serviceName, interfaceList] : serviceMap)
+    {
+        bool cpuInterface = false;
+        bool associationInterface = false;
+        for (const auto& interface : interfaceList)
+        {
+            if (interface == "xyz.openbmc_project.Inventory.Item.Cpu")
+            {
+                cpuInterface = true;
+            }
+            else if (interface == "xyz.openbmc_project.Association.Definitions")
+            {
+                associationInterface = true;
+            }
+        }
+
+        if (cpuInterface && associationInterface)
+        {
+            setLocationIndicatorActive(asyncResp, objectPath,
+                                       locationIndicatorActive);
+        }
+    }
+}
+
+inline void afterSetProcessorObject(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, bool locationIndicatorActive,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    for (const auto& [objectPath, serviceMap] : subtree)
+    {
+        // Ignore any objects which don't end with our desired cpu name
+        sdbusplus::message::object_path path(objectPath);
+        std::string name = path.filename();
+        if (name.empty() || name != processorId)
```

Comment:
```
if (name != processorId)
```

Reply:
Agreed, I went ahead and made this change.